### PR TITLE
Unit test support (CMake and BigInt)

### LINKED ISSUE #108: BigInt failing when reassigning it to char*
When adding tests for BigInt some bugs occurred when reassigning values as strings, I also tested constructors and other assignments types and they were fine. It looks like it's just prepending to the old input.

Am I doing something wrong? If needed, here is the [Full code link](https://pastebin.com/X3dJKpN7) with the helper implementation and all other test cases.

Test cases:
```
BOOST_AUTO_TEST_CASE(string_assignment_ok_1) {
  BigIntHelper<int> obj(-255);
  obj.obj = "12345678987654321234567890";
  obj.check_value("12345678987654321234567890");
}
// Fails with [12345678987654321234567890 != -25512345678987654321234567890]

BOOST_AUTO_TEST_CASE(string_assignment_ok_2) {
  BigIntHelper<int> obj(-255);
  obj.obj = "255";
  obj.check_value("255");
}
// Fails with [255 != -255255] 

BOOST_AUTO_TEST_CASE(string_assignment_ok_3) {
  BigIntHelper<int> obj(255);
  obj.obj = "-255";
  obj.check_value("-255");
}
// Fails with [-255 != -255255]

BOOST_AUTO_TEST_CASE(string_assignment_ok_4) {
  const BigInt::onedig_t base = 10;
  const char *input = "12345678987654321234567890";
  BigIntHelper<char> obj(input, base);
  obj.obj = input;
  obj.check_value(input);
}
// Fails with [12345678987654321234567890 != 1234567898765432123456789012345678987654321234567890]
```

### LINKED ISSUE #110: BigInt does not handle signed comparations
Compare method is not behaving well for BigInt objects initialized with a negative value. 

Example of failure:
```
BigInt obj(-255);
int expected = -255 > -200;
int actual = obj > -200;
assert(expected == actual); // This fails!
```

If using only unsigned values, it works:
```
BigInt obj(255);
int expected = 255 > 200;
int actual = obj > 200;
assert(expected == actual); // OK!
```

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
This should add the base template for Unit test integration with CMake, Boost.Test, Fuzzers and Sanitizers.

Also, it should fix:
- Closes https://github.com/esbmc/esbmc/issues/108
- Closes https://github.com/esbmc/esbmc/issues/110

### CODE REVIEW

File: src/big-int/bigint.cpp [hunk 1]
```diff
@@ -466,6 +466,9 @@ BigInt &BigInt::operator=(BigInt const &y)
 
 BigInt &BigInt::operator=(char const *s)
 {
+  reallocate(small);
+  length = 0;
+  positive = true;
```
Comment (mikhailramalho) [inline src/big-int/bigint.cpp:471]: Why is this needed?

Comment (rafaelsamenezes) [reply src/big-int/bigint.cpp:471]: This intends to fix the problems from Issue #108

When reassigning from a string `scan_on` assumes that the `length` is already set. Specifics:

- `reallocate` to clean old data. This should fix replacing with a smaller number.
- `length` is set to 0 to start from the beginning to parse the number correctly
- `positive` is set to true because the `scan` function will only check if the number is negative with the "-" sign. So, if the older number was negative and the newer is positive BigInt will think that the number is negative.
